scipy/fftpack/helper.py

```diff
@@ -151,3 +152,126 @@ def next_fast_len(target):
     if p5 < match:
         match = p5
     return match
+
+
+def _init_nd_shape_and_axes(x, shape, axes):
+    """Handle shape and axes arguments for n-dimensional transforms.
+
+    Returns the shape and axes in a standard form, taking into account negative
+    values and checking for various potential errors.
+
+    Parameters
+    ----------
+    x : array_like
+        The input array.
+    shape : int or array_like of ints or None
+        The shape of the result.  If both `shape` and `axes` (see below) are
+        None, `shape` is ``x.shape``; if `shape` is None but `axes` is
+        not None, then `shape` is ``scipy.take(x.shape, axes, axis=0)``.
+        If `shape` is -1, the size of the corresponding dimension of `x` is
+        used.
+    axes : int or array_like of ints or None
+        Axes along which the calculation is computed.
+        The default is over all axes.
+        Negative indices are automatically converted to their positive
+        counterpart.
+
+    Returns
+    -------
+    shape : array
+        The shape of the result. It is a 1D integer array.
+    axes : array
+        The shape of the result. It is a 1D integer array.
+
+    """
+    x = asarray(x)
+    noshape = shape is None
+    noaxes = axes is None
+
+    if noaxes:
+        axes = arange(x.ndim, dtype=intc)
+    else:
+        axes = atleast_1d(axes)
+
+    if axes.size == 0:
+        axes = axes.astype(intc)
+
+    if not axes.ndim == 1:
+        raise ValueError("when given,  axes values must be a scalar or vector")
+    if not issubdtype(axes.dtype, integer):
+        raise ValueError("when given, axes values must be integers")
+
+    axes = where(axes < 0, axes + x.ndim, axes)
+
+    if axes.size != 0 and axes.max() >= x.ndim:
+        raise ValueError("axes exceeds dimensionality of input")
```

Comment:
you should also check for too much negativity, which at this point in the code I think is `axes.min() < 0`

Reply:
Done, with a unit test added for it.